src/constructs/iam/policies/kcl.test.ts

```diff
@@ -1,12 +1,21 @@
 import { Template } from "aws-cdk-lib/assertions";
+import { Stream } from "aws-cdk-lib/aws-kinesis";
 import { attachPolicyToTestRole, simpleGuStackForTesting } from "../../../utils/test";
 import { GuKCLPolicy } from "./kcl";
 
 describe("GuKCLPolicy", () => {
   it("should create a policy granting sufficient permissions for the KCL", () => {
     const stack = simpleGuStackForTesting();
 
-    const policy = new GuKCLPolicy(stack, { streamName: "streamFoo", applicationName: "appBar" });
+    const streamName: string = "streamFoo";
+    const policy = new GuKCLPolicy(stack, {
+      stream: Stream.fromStreamArn(
+        stack,
+        streamName,
+        `arn:aws:kinesis:${stack.region}:${stack.account}:stream/${streamName}`,
+      ),
+      applicationName: "appBar",
+    });
```

Comment:
Although this passes tests here in `guardian/cdk`, it fails when used in https://github.com/guardian/apple-news/pull/428 - see https://github.com/guardian/apple-news/pull/428/files#r1955995331